main.go

```diff
@@ -207,6 +209,28 @@ func populateArgs(c *cli.Context) (
 	return
 }
 
+func callListRecursive(mountPoint string) (err error) {
+	logger.Debugf("Started recursive metadata-prefetch of directory: \"%s\" ...", mountPoint)
+	numItems := 0
+	err = filepath.WalkDir(mountPoint, func(path string, d fs.DirEntry, err error) error {
+		//logger.Infof("Walked path:%s, d=%s, isDir=%v", path, d.Name(), d.IsDir())
+		if err != nil {
```

Comment:
> How should the error be handled? IIUC, the behavior implemented here is to stop the walking entirely if an error is detected. Is that what we want?

Yes, the intent is to fail this function of walking fails even on one directory/file.

> If so, then we should log that the walking failed

It is logged by the caller, which receives and logs the error, dependending on the caller's situation.

> If we are okay to skip it and continue walking other directories, fs.SkipDir should be returned.

Skipping failed walk and silently ignoring it will defeat the purpose of the walk. In case of asynchronous mode, the failed walk generates an error log, and prints it into the logfile. In case of synchronous mode, the failed walk causes the mount itself to fail.